src/router.cpp

```diff
@@ -139,6 +139,15 @@ int zmq::router_t::xsetsockopt (int option_,
             }
             break;
 
+#ifdef ZMQ_BUILD_DRAFT_API
+        case ZMQ_ROUTER_NOTIFY:
+            if (is_int && value >= 0 && value <= 3) {
```

Comment:
Use ZMQ_NOTIFY_CONNECT | ZMQ_NOTIFY_DISCONNECT rather than the "magic" literal 3 here.